src/ui/osx/TogglDesktop/Feature/TouchBar/TouchBar+PrivateAPIs.m

```diff
@@ -0,0 +1,86 @@
+//
+//  TouchBar+PrivateAPIs.m
+//  TogglDesktop
+//
+//  Created by Nghia Tran on 11/22/19.
+//  Copyright © 2019 Alari. All rights reserved.
+//
+
+#import "TouchBar+PrivateAPIs.h"
+
+@interface NSTouchBar ()
+/* macOS 10.13 */
++ (void)presentSystemModalFunctionBar:(NSTouchBar *)touchBar
+                            placement:(long long)placement
+             systemTrayItemIdentifier:(NSTouchBarItemIdentifier)identifier;
++ (void)presentSystemModalFunctionBar:(NSTouchBar *)touchBar
+             systemTrayItemIdentifier:(NSTouchBarItemIdentifier)identifier;
++ (void)dismissSystemModalFunctionBar:(NSTouchBar *)touchBar;
++ (void)minimizeSystemModalFunctionBar:(NSTouchBar *)touchBar;
+
+/* macOS 10.14 */
++ (void)presentSystemModalTouchBar:(NSTouchBar *)touchBar
```

Comment:
Those new Private API will keep the TimeEntry Touch bar persistently, regardless of controls, which is a first responder.

This approach could fix the known-issues that the NSTextField (allowseditingtextattributes=true) doesn't override the Timeline Touch Bar anymore. However, we have to manually minimize or dismiss when the app is go to background / foreground.